stl/inc/atomic

```diff
@@ -1069,35 +1127,38 @@ struct _Atomic_storage<_Ty, 8> { // lock-free using 8-byte intrinsics
     }
 #endif // _HAS_CXX20
 
-    _Atomic_padded<_Ty> _Storage;
+    typename _Atomic_storage_types<_Ty>::_TStorage _Storage;
 };
 
-#if 0 // TRANSITION, ABI
-#if defined(_M_X64) || defined(_M_ARM64)
+#ifdef _WIN64
 template <class _Ty>
-struct _Atomic_storage<_Ty, 16> { // lock-free using 16-byte intrinsics
+struct _Atomic_storage<_Ty&, 16> { // lock-free using 16-byte intrinsics
+    // TRANSITION, ABI: replace '_Ty&' with '_Ty' everywhere in this specialization
+
+    using _TVal = remove_reference_t<_Ty&>;
+
     _Atomic_storage() = default;
 
-    /* implicit */ constexpr _Atomic_storage(const _Ty _Value) noexcept
+    /* implicit */ constexpr _Atomic_storage(conditional_t<is_reference_v<_Ty&>, _Ty&, const _TVal> _Value) noexcept

```

Comment:
`is_reference_v<_Ty&>` is always true. I *think* this is intentional, given the TRANSITION comment, just making sure.

Reply:
Made `TRANSITION` another way around, so that this part does not stand out

Reply:
Had to keep this way, hopefully TRANSITION comment will work